### DISCUSSION TITLE
Getting an error when uploading new in media library and saving changes

### DISCUSSION THREAD

**Discussion by l-mtc** [awcodes/filament-curator#352] [Q&A]
Filament\Forms\Components\BaseFileUpload::Filament\Forms\Components\{closure}(): Argument #1 ($file) must be of type Livewire\Features\SupportFileUploads\TemporaryUploadedFile|string, array given

**Comment by awcodes** [comment]
I need more information as to what you are trying to do.

**Reply by l-mtc** [reply]
When I go into the media library and edit an existing image then hit save I get this error, I'm storing it in s3

**Reply by awcodes** [reply]
When you say 'media library' are you referring to the Edit Media page or the panel in the modal when using the CuratorPicker field?

**Reply by l-mtc** [reply]
Edit media page

**Comment by awcodes** [comment]
Hmm. I can't seem to replicate the issue. uploading new media to an existing media item is working fine for me.

Can you share a video of what it happening, so I can understand better exactly what you are doing that is causing the issue?

**Comment by l-mtc** [comment]
No worries, I've uploaded the video here, can you let me know when you've downloaded so I can delete the link: https://share.1password.com/s#mt5wpq1u3KLvueGRDJTdVts4VdbqmWeZOchIIHwItBg

**Reply by awcodes** [reply]
i've got it. you can delete it now. thanks.

**Comment by l-mtc** [comment]
I did try before I recorded this and if I hit the save at the bottom it says saved, then if I click save at the top it errors

**Comment by awcodes** [comment]
hmm. i'm not sure what's going on, but the error you're getting is from Filament's BaseFileUpload class.

https://github.com/awcodes/filament-curator/assets/3596800/8e7f7ccf-067a-4725-8f72-edae83e7052e
